sno/working_copy/base.py

```diff
@@ -53,25 +57,26 @@ def STATE_TABLE(self):
         return self._sno_table(self.STATE_NAME)
 
     @classmethod
-    def get(cls, repo, init_if_missing=False):
+    def get(cls, repo, allow_uncreated=False, allow_invalid_state=False):
         """
         Get the working copy associaated with this sno repo.
         If no working copy is present, returns None -
-        unless init_if_missing is returned, in which case it returns a WorkingCopy object that is not "created".
+        unless allow_uncreated is returned, in which case it returns a WorkingCopy object that is not "created".
         The type of this WorkingCopy object depends on the repo config - it represents a particular type of WorkingCopy
         at a particular location, which nevertheless doesn't exist yet. It is up to the caller to create it by calling
         `create()` and to populate it by calling eg `write_full()`.
 
         """
-        if init_if_missing:
-            cls.ensure_config_exists(repo)
-
         repo_cfg = repo.config
-        path_key = SNO_WORKINGCOPY_PATH
+        path_key = cls.SNO_WORKINGCOPY_PATH
         if path_key not in repo_cfg:
             return None
 
         path = repo_cfg[path_key]
+        return cls.get_at_path(repo, path, allow_uncreated, allow_invalid_state)
+
+    @classmethod
+    def get_at_path(cls, repo, path, allow_uncreated=False, allow_invalid_state=False):
```

Comment:
```suggestion
    def get_at_path(cls, repo, path, *, allow_uncreated=False, allow_invalid_state=False):
```
(and change caller(s) to suit)

multiple boolean flags are a good fit for keyword-only arguments since it's too easy to pass them in the wrong order otherwise

Reply:
Done